sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala

```diff
@@ -2011,7 +2011,7 @@ private[sql] object QueryExecutionErrors extends QueryErrorsBase {
 
   def batchMetadataFileNotFoundError(batchMetadataFile: Path): SparkFileNotFoundException = {
     new SparkFileNotFoundException(
-      errorClass = "_LEGACY_ERROR_TEMP_2206",
+      errorClass = "INTERNAL_ERROR_BATCH",
```

Comment:
It's user facing exception - this can happen in both sides, engine has a bug or users did the crazy thing (try to adjust the checkpoint), or even some lifecycle policy of the storage (deleted by retention policy, I meant, accidentally).